pilot/cmd/pilot-agent/main.go

```diff
@@ -290,6 +292,16 @@ var (
 				defer stsServer.Stop()
 			}
 
+			// Start a local DNS server on 15053, forwarding to DNS-over-TLS server
+			// This will not have any impact on app unless interception is enabled.
+			// We can't start on 53 - istio-agent runs as user istio-proxy.
+			// This is available to apps even if interception is not enabled.
+
+			// TODO: replace hardcoded list with settings.
+			dnsSrv := dns.InitDNSAgent(dns.DNSUp.Get(), sa.RootCert,
```

Comment:
Most likely - I think we should only enable if strict mode is enabled ( since in strict mode the assumption is security has priority, and DNS should be secured). 

But I would leave it on for at least few days, to get more coverage.

Reply:
Added explicit code - if the setting is empty will not init.